### ISSUE TITLE
[BUG] script xtensa-build-zephyr.py fails on python 3.6

### ISSUE THREAD

**Issue by aborisovich** [thesofproject/sof#6121]
**Describe the bug**
Running script `./scripts/xtensa-build-zephyr.py tgl` on Python 3.6.X will result in error since `shlex.join`
was introduced in Python 3.8:
![image](https://user-images.githubusercontent.com/12482515/184032537-285922c4-d85e-43a2-af98-d81b74363098.png)

**To Reproduce**
Install Python 3.6.X according to instructions on [Zephyr - Getting Started](https://docs.zephyrproject.org/latest/develop/getting_started/index.html) guide.
Follow build instructions defined in SOF docs [Build with Zephyr](https://thesofproject.github.io/latest/getting_started/build-guide/build-with-zephyr.html) and use xtensa-build-zephyr.py script.

**Reproduction Rate**
Always.

**Expected behavior**
SOF project documentation contains information that Python 3.8.X+ is required or
script is fixed and working with declared by Zephyr Python 3,6.X version.

**Impact**
Annoyance and confusion.

**Environment**
Where bug occurs:
    * SOF: main
OS: Windows, Linux
Bug introduced in commit ce8bc4a3941a5f005427caf93dea32d996161ef2

**Comment by nashif** [comment]
> Install Python 3.6.X according to instructions on [Zephyr - Getting Started](https://docs.zephyrproject.org/latest/develop/getting_started/index.html) guide.

that is the minimum required, you do not need to use it, you should use a more recent version if you can.

**Comment by aborisovich** [comment]
@lgirdwood @marc-hb can we define Python 3.8.x as minimal requirement for SOF project?
Which Python version CI runs (for both Zephyr and non-Zephyr CI)?

**Comment by lgirdwood** [comment]
> @lgirdwood @marc-hb can we define Python 3.8.x as minimal requirement for SOF project? Which Python version CI runs (for both Zephyr and non-Zephyr CI)?

Ack - I think most distros are running 3.10 now.

**Comment by marc-hb** [comment]
tl;dr: 3.8 requirement is fine by me.

Python 3.6 is out of support, should never be used: https://www.python.org/downloads

Zephyr requirements are one thing, SOF requirements are another. Maybe Zephyr wants to be extraordinarily nice and not break backward-compatibility with insecure 3.6 but that does not mean SOF has to be as "nice". Zephyr has obviously many more users than SOF.

3.7 is supported for only one more year.

3.8 was released in 2019, I think that's plenty enough time ago.

**Comment by aborisovich** [comment]
Ok then, is anyone willing to update SOF project documentation with this information?
I think it should be placed somewhere in the https://thesofproject.github.io/latest/getting_started/index.html#build-sof .

**Comment by lgirdwood** [comment]
@marc-hb could you update sof-docs with Python version and the new alsa-lib version (merged) ?

**Comment by marc-hb** [comment]
Python >= 3.8 assert submitted in #6162. Assert is better than documentation: documentation gets out of date and people don't read it (gross exaggeration to get the point across).

For alsa-lib sof-docs has always been recommending to build from git (but many people skip that)
